rest_framework/utils/mediatypes.py

```diff
@@ -83,5 +83,5 @@ def precedence(self):
     def __str__(self):
         ret = "%s/%s" % (self.main_type, self.sub_type)
         for key, val in self.params.items():
-            ret += "; %s=%s" % (key, val)
+            ret += "; %s=%s" % (key, val.decode('ascii'))
```

Comment:
I still don't get why this is required. I'm afraid we create a regression here.

Reply:
The `%s` placeholder implies a call to `str()` before substitution. In Python3, this looks like:

```py
>>> >>> print('%s' % b'foo')
b'foo'
```

So, the end result of the formatted string is something like `"test/*; foo=b'bar'"`. Notice the extra characters `b''`. My understanding is this repr bytes value was not intended as part of the return value. If truee, this wasn't previously correct. Enabling BytesWarning caught this bug. This was caught by fixing the test `test_mediatype_string_representation()`.

Reply:
@jdufresne thanks for the feedback. Indeed, you're correct. Thanks for the nice work !